src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -665,6 +666,11 @@ UCS_CLASS_INIT_FUNC(uct_rc_mlx5_iface_common_t,
         goto cleanup_tm;
     }
 
+    if ((self->super.config.fence_mode == UCT_RC_FENCE_MODE_NONE) &&
```

Comment:
unite uct_ib_device_has_pci_atomics() and md->super.relaxed_order to
uct_ib_iface_need_fence()